tiatoolbox/models/classification/patch_predictor.py

```diff
@@ -206,20 +202,107 @@ def __init__(
 
+        self._iostate = iostate  # for storing original
+        self.iostate = None  # for runtime
+        self.model = model  # for runtime, such as after wrapping with nn.DataParallel
+        self.pretrained_model = pretrained_model
         self.batch_size = batch_size
         self.num_loader_worker = num_loader_worker
         self.verbose = verbose
 
     @staticmethod
-    def __postprocess(probabilities):
+    def merge_predictions(
+        img,
+        output,
+        resolution=None,
+        units=None,
+    ):
+        """Merge patch-level predictions to form a 2-dimensional prediction map.
+
+        Args:
+            img (:obj:`str` or :obj:`pathlib.Path` or :class:`numpy.ndarray`):
+                A HWC image or a path to WSI.
+            output_list (list): List of outputs generated by the model.
+            resolution (float): Resolution of merged predictions.
+            units (str): Units of resolution used when merging predictions. This
+                must be the same `units` used when processing the data.
+
+        Returns:
+            prediction_map (ndarray): Merged predictions as a 2D array.
+            overlay (ndarray): Overlaid output if return_overlay is set to True.
+
+        """
+        reader = get_wsireader(img)
+        if isinstance(reader, VirtualWSIReader):
+            warnings.warn(
+                " ".join(
+                    [
+                        "Image is not pyramidal hence read is forced to be",
+                        "at `units='baseline'` and `resolution=1.0`.",
+                    ]
+                )
+            )
+            resolution = 1.0
+            units = "baseline"
+
+        canvas_shape = reader.slide_dimensions(resolution=resolution, units=units)
+        canvas_shape = canvas_shape[::-1]  # XY to YX
+
+        # may crash here, do we need to deal with this ?
+        output_shape = reader.slide_dimensions(
+            resolution=output["resolution"], units=output["units"]
+        )
+        output_shape = output_shape[::-1]  # XY to YX
+        fx = np.array(canvas_shape) / np.array(output_shape)
+
+        if "probabilities" not in output.keys():
+            coordinates = output["coordinates"]
+            predictions = output["predictions"]
+            denominator = None
+            output = np.zeros(list(canvas_shape), dtype=np.float32)
+        else:
+            coordinates = output["coordinates"]
+            predictions = output["probabilities"]
+            num_class = np.array(predictions[0]).shape[0]
+            denominator = np.zeros(canvas_shape)
+            output = np.zeros(list(canvas_shape) + [num_class], dtype=np.float32)
+
+        for idx, bound in enumerate(coordinates):
+            prediction = predictions[idx]
+            # assumed to be in XY
+            # top-left for output placement
+            tl = (np.array(bound[:2]) * fx + 0.5).astype(np.int32)
```

Comment:
doesn't `np.floor()` do the same functionality? if yes, it might be better to consider rewriting this line using `np.floor` to keep the code easier to understand.